src/Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.cpp

```diff
@@ -0,0 +1,64 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/GeneralizedHarmonic/BoundaryConditions/BjorhusImpl.hpp"
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace GeneralizedHarmonic::BoundaryConditions::Bjorhus {
+template <size_t VolumeDim, typename DataType>
+void add_constraint_preserving_terms_to_dt_v_psi(
+    const gsl::not_null<tnsr::aa<DataType, VolumeDim, Frame::Inertial>*>
+        bc_dt_v_psi,
+    const tnsr::I<DataType, VolumeDim, Frame::Inertial>&
+        unit_interface_normal_vector,
+    const tnsr::iaa<DataType, VolumeDim, Frame::Inertial>&
+        three_index_constraint,
+    const std::array<DataType, 4>& char_speeds) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*bc_dt_v_psi)) !=
+               get_size(get<0>(unit_interface_normal_vector)))) {
+    *bc_dt_v_psi =
```

Comment:
since this function is named `add` it shouldn't be setting the input to zero. Can you add a test that checks that if `bc_dt_v_psi` is initialized to `1.0` that the output is `1.0 + correction`?

Reply:
I think this can also be done with the pypp test using the `initial_result_values` argument (documentation here: https://spectre-code.org/namespacepypp.html#aba1070d7852ff195f81a101605de88d3)

Reply:
You're right, but the correct fix is to actually rename the function. The prefix `add_` is not relevant now since the function only computes the correction terms. I've renamed the function to `constraint_preserving_corrections_dt_v_psi`, and changed the function too to not `add` to the pre-stored value, but simply assign, as it should have to begin with.